deepchecks/vision/checks/performance/segment_performance.py

```diff
@@ -0,0 +1,240 @@
+        else:
+            if len(image_properties) == 0:
+                raise DeepchecksValueError('image_properties list cannot be empty')
+
+            properties_by_name = {p for p in image_properties if isinstance(p, str)}
+            unknown_properties = properties_by_name.difference(ImageFormatter.IMAGE_PROPERTIES)
+
+            if len(unknown_properties) > 0:
+                raise DeepchecksValueError(
+                    'received list of unknown image properties '
+                    f'- {sorted(unknown_properties)}'
+                )
+
+            self.image_properties = image_properties
+
+        self.alternative_metrics = alternative_metrics
+        self.number_of_bins = number_of_bins
+        self.number_of_samples_to_infer_bins = number_of_samples_to_infer_bins
+        self._state = None
+
+    def initialize_run(self, context: Context, dataset_kind: DatasetKind):
+        """Initialize run before starting updating on batches"""
+        # First we will aggregate samples up to defined amount (number_of_samples_to_infer_bins), when we reach
+        # the amount we will define the bins and populate them
+        self._state = {'samples_for_binning': [], 'bins': None}
+        # Initialize image properties. Doing this not in the init because we use the dataset object.
+        dataset = context.get_data_by_kind(dataset_kind)
+        string_props = {p: getattr(dataset.image_formatter, p) for p in self.image_properties if isinstance(p, str)}
+        func_props = {p.__name__: p for p in self.image_properties if callable(p)}
+        self._state['properties_functions'] = {**string_props, **func_props}
+
+    def update(self, context: Context, batch: t.Any, dataset_kind: DatasetKind):
+        dataset = context.get_data_by_kind(dataset_kind)
+        images = dataset.image_formatter(batch)
+        predictions = context.infer(batch)
+        labels = dataset.label_formatter(batch)
+
+        samples_for_bin: t.List = self._state['samples_for_binning']
+        bins = self._state['bins']
+        properties_functions: t.Dict = self._state['properties_functions']
+
+        # Initialize a list of all properties per image sample
+        batch_properties = []
+        for _ in range(len(images)):
+            batch_properties.append({})
+        for prop_name, func in properties_functions.items():
+            for index, image_result in enumerate(func(images)):
+                batch_properties[index][prop_name] = image_result
+
+        batch_data = zip(labels, predictions, batch_properties)
+        # If we already defined bins, add the current data to them
+        if bins is not None:
+            _divide_to_bins(bins, batch_data)
+        else:
+            # Add the current data to the samples list
+            samples_for_bin.extend(batch_data)
+            # Check if enough data to infer bins
+            if len(samples_for_bin) < self.number_of_samples_to_infer_bins:
```

Comment:
```suggestion
            if len(samples_for_bin) >= self.number_of_samples_to_infer_bins:
```
We want more sample then that threshold right?

Reply:
fixed